src/search-params/search-params-manager.js

```diff
@@ -0,0 +1,127 @@
+import React, {useEffect} from 'react'
+import { useDispatch } from 'react-redux'
+import qs from 'qs'
+import _ from 'lodash'
+import { connect } from "react-redux"
+import { datasetSearch } from "../store/actions.js"
+import PropTypes from 'prop-types'
+
+const defaults = {
+  apiAccessible: true,
+  sortOrder: 'name_asc',
+  page: 1,
+  searchText: '',
+  facets: {}
+}
+
+class SearchParamsManager {
+  constructor(history) {
+    this.pushHistory = history.push
+    this.params = qs.parse(history.location.search, { ignoreQueryPrefix: true })
+
+    this.getParam = this.getParam.bind(this)
+    this.getParams = this.getParams.bind(this)
+
+    this.toggleApiAccessible = this.toggleApiAccessible.bind(this)
+    this.toggleFacet = this.toggleFacet.bind(this)
+    this.updateSearchText = this.updateSearchText.bind(this)
+    this.updateSortOrder = this.updateSortOrder.bind(this)
+    this.updatePage = this.updatePage.bind(this)
+
+    this.updateParams = this.updateParams.bind(this)
+  }
+
+  getParams() {
+    return {
+      apiAccessible: this.apiAccessible,
+      page: this.page,
+      facets: this.facets,
+      searchText: this.searchText,
+      sortOrder: this.sortOrder
+    }
+  }
+
+  toggleApiAccessible() {
+    const updatedApiAccessibleFlag = ! this.apiAccessible
+
+    this.updateParams({ apiAccessible: updatedApiAccessibleFlag, page: 1 })
+  }
+
+  toggleFacet(name, value) {
+    const facetValues = this.facets[name]
+    const updatedFacets = Object.assign({}, this.facets, { [name]: _.xor(facetValues, [value]) })
+
+    this.updateParams({ facets: updatedFacets, page: 1 })
+  }
+
+  updateSortOrder(sort) {
+    this.updateParams({ sort: sort })
+  }
+
+  updateSearchText(searchText) {
+    this.updateParams({ q: searchText, page: 1 })
+  }
+
+  updatePage(page) {
+    this.updateParams({ page: page })
+  }
+
+  updateParams(params) {
+    const updatedSearch = Object.assign({}, this.params, params)
+    const updatedSearchEncoded = qs.stringify(
+      updatedSearch,
+      { arrayFormat: 'brackets', addQueryPrefix: false }
+    )
+
+    this.pushHistory({search: updatedSearchEncoded})
+  }
+
+  get apiAccessible() {
+    return this.getParam('apiAccessible') !== 'false'
+  }
+
+  get sortOrder() {
+    return this.getParam('sort') || defaults.sortOrder
+  }
+
+  get page() {
+    return Number.parseInt(this.getParam('page')) || defaults.page
```

Comment:
Could we parse this on initial and on update so that we don't have to parse / check the default every time we get it?